ProcessLib/SmallDeformationWithLIE/Common/PostUtils.cpp

```diff
@@ -0,0 +1,205 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#include "PostUtils.h"
+
+#include <map>
+#include <vector>
+
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/MeshEditing/DuplicateMeshComponents.h"
+
+namespace ProcessLib
+{
+namespace SmallDeformationWithLIE
+{
+
+namespace
+{
+
+template <typename T>
+inline void sort_unique(std::vector<T> &vec)
+{
+    std::sort(vec.begin(), vec.end());
+    vec.erase(std::unique(vec.begin(), vec.end()), vec.end());
+}
+
+} // no named namespace
+
+PostProcessTool::PostProcessTool(
+    MeshLib::Mesh const& org_mesh,
+    std::vector<MeshLib::Node*> const& vec_fracture_nodes,
+    std::vector<MeshLib::Element*> const& vec_fracutre_matrix_elements)
+    :_org_mesh(org_mesh)
+{
+    if (!org_mesh.getProperties().hasPropertyVector("displacement")
+        || !org_mesh.getProperties().hasPropertyVector("displacement_jump1")
+        || !org_mesh.getProperties().hasPropertyVector("levelset1")
+        )
+    {
+        OGS_FATAL("The given mesh does not have relevant properties");
+    }
+
+    // clone nodes and elements
+    std::vector<MeshLib::Node*> new_nodes(MeshLib::copyNodeVector(org_mesh.getNodes()));
+    std::vector<MeshLib::Element*> new_eles(
+        MeshLib::copyElementVector(org_mesh.getElements(), new_nodes));
+
+    // duplicate fracture nodes
+    for (auto const* org_node : vec_fracture_nodes)
+    {
+        auto duplicated_node = new MeshLib::Node(org_node->getCoords(), new_nodes.size());
+        new_nodes.push_back(duplicated_node);
+        _map_dup_newNodeIDs[org_node->getID()] = duplicated_node->getID();
+    }
+
+    // split elements using the new duplicated nodes
+    auto prop_levelset = org_mesh.getProperties().getPropertyVector<double>("levelset1");
+    for (auto const* org_e : vec_fracutre_matrix_elements)
+    {
+        // only matrix elements
+        if (org_e->getDimension() != org_mesh.getDimension())
+            continue;
+
+        auto const eid = org_e->getID();
+
+        // keep original if the element has levelset=0
+        if ((*prop_levelset)[eid] == 0)
```

Comment:
You are comparing double values here, aren't you? Is it safe to use strict equality?